backend/app/Repositories/EloquentChartVisitorsRepository.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Repositories;
+
+use App\Actions\Visitors\GetNewChartVisitorsByDateRangeRequest;
+use App\Repositories\Contracts\ChartVisitorsRepository;
+use Carbon\Carbon;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+
+class EloquentChartVisitorsRepository implements ChartVisitorsRepository
+{
+    public function getNewVisitorsByDate(GetNewChartVisitorsByDateRangeRequest $request): Collection
+    {
+        $userId = Auth::user()->id;
+        $startData = Carbon::createFromTimestampUTC($request->getStartDate())->toDateTimeString();
+        $endData = Carbon::createFromTimestampUTC($request->getEndDate())->toDateTimeString();
+
+        $response = DB::select('SELECT COUNT(*), period from
```

Comment:
I've already written it to Mariia, you can write raw sql more readable and more functional

```php
<?php

function toTimestamp(string $columnName): string
{
    return "extract(epoch FROM $columnName)";
}

function toInteger(string $expression): string
{
    return "CAST ($expression) AS INTEGER";
}

function roundDate(string $columnName, int $period): string
{
    return $this->toTimestamp($columnName) . " - MOD(" . $this->toInteger($this->toTimestamp($columnName)) . ", " . $period . ")";
}

$subQuery = "SELECT visits.*, (" . $this->roundDate('created_at', $interval) . ") as date" .
    "FROM visits" . 
    "WHERE " . $this->toTimestamp('created_at') . " >= :start_date" .
    "AND" .
    $this->toTimestamp('created_at') . " <= :end_date";
$query = DB::raw("SELECT COUNT(*) as visits, date FROM ($subQuery) AS periods GROUP BY date");

$result =  DB::select((string)$query, [
    'start_date' => $filterData->getStartDate()->getTimestamp(),
    'end_date' => $filterData->getEndDate()->getTimestamp(),
]);
```